### PR TITLE
flash size: +2M(no spiffs)

### PR DESCRIPTION
**Pull request by d-a-v**
additionally:
board generator: comments and simplify
ldscripts: link error when trying to use spiffs when unavailable

exemple with 512k no spiffs and a sketch using spiffs:
```
/tmp/arduino_build_788324/arduino.ar(Updater.cpp.o):(.text._ZN12UpdaterClass5beginEji+0xc): undefined reference to `_SPIFFS_start'
/tmp/arduino_build_788324/arduino.ar(spiffs_api.cpp.o): In function `SPIFFSImpl::format()':
.../hardware/esp8266com/esp8266/cores/esp8266/spiffs_api.h:156: undefined reference to `_SPIFFS_end'
.../hardware/esp8266com/esp8266/cores/esp8266/spiffs_api.h:161: undefined reference to `_SPIFFS_page'
.../hardware/esp8266com/esp8266/cores/esp8266/spiffs_api.h:162: undefined reference to `_SPIFFS_block'
collect2: error: ld returned 1 exit status
```
This behavior is opened to discussion.
I think this could reduce administrative time regarding misconfiguration.

### CODE REVIEW

File: tools/boards.txt.py [hunk 1]
```diff
@@ -1153,24 +1155,31 @@ def flash_size (size_bytes, display, optname, ld, desc, max_upload_size, spiffs_
...
+    f1m.update( flash_size(0x100000,   '1M', '1m64',     '64K SPIFFS',  958448,  0xEB000,   0x10000, 4096))
+    f1m.update( flash_size(0x100000,   '1M', '1m128',   '128K SPIFFS',  892912,  0xDB000,   0x20000, 4096))
+    f1m.update( flash_size(0x100000,   '1M', '1m144',   '144K SPIFFS',  876528,  0xD7000,   0x24000, 4096))
+    f1m.update( flash_size(0x100000,   '1M', '1m160',   '160K SPIFFS',  860144,  0xD3000,   0x28000, 4096))
+    f1m.update( flash_size(0x100000,   '1M', '1m192',   '192K SPIFFS',  827376,  0xCB000,   0x30000, 4096))
+    f1m.update( flash_size(0x100000,   '1M', '1m256',   '256K SPIFFS',  761840,  0xBB000,   0x40000, 4096))
+    f1m.update( flash_size(0x100000,   '1M', '1m512',   '512K SPIFFS',  499696,  0x7B000,   0x80000, 8192))
+    f2m =       flash_size(0x200000,   '2M', '2m0',      'no SPIFFS',  1044464, 0x100000)
+    f2m.update( flash_size(0x200000,   '2M', '2m',       '1M SPIFFS',  1044464, 0x100000,   0xFB000, 8192))
+    f4m =       flash_size(0x400000,   '4M', '4m1m',     '1M SPIFFS',  1044464, 0x300000,   0xFB000, 8192)
```
Comment (devyte) [inline tools/boards.txt.py]: I suggest a 4M0, at least for testing purposes if nothing else. I don't know if it's the case for everyone else, but I have a whole lot of boards and they're all 4MB flash.

**Comment by devyte** [comment]
@d-a-v this has conflicts now, could you please resolve?

File: tools/boards.txt.py (hunk 1 above)
Comment (d-a-v) [reply tools/boards.txt.py]: added in #5103

**Comment by d-a-v** [comment]
replaced by #5103

**Pull request closed without merging**